### DISCUSSION TITLE
FlxText XY discrepancy

### DISCUSSION THREAD

**Discussion by DigiEggz** [HaxeFlixel/flixel#2528] [Q&A] [answered]
I'm running the latest HaxeFlixel and openfl and have noticed a strange issue:

![ani](https://user-images.githubusercontent.com/3136062/162236295-dbdec418-db48-4310-8215-2e9bc6b21e9f.gif)

There's always difference of +1px on the X and Y axis in the placement of all FlxText between desktop and web versions. This is particularly noticeable on buttons. The first frame above shows the desktop vs web versions. I've tested Windows, Neko and macOS and they all behave the same. I've also tested different browsers to see if it was a text rendering difference but they display the same as well.

This is also noticeable if you create a FlxText at 0, 0 and compare the difference in placement between desktop and web. For me, there's a 1x1 pixel (down and right) gap between the versions.

I'm not sure if this is caused by something within my settings or not, but any confirmation or help is appreciated. @Geokureli @Gama11 may have an idea since they're more familiar with the inner workings.

**Comment by Geokureli** [comment]
tbh, I'm not sure. My first guess is that this is a openfl/lime issue, since FlxText is drawn from an openfl textfield. Are you sure it's not a problem with the Button centering it's label? if you draw a FlxText at 0,0 does it look the same on both targets? what about centering it, what width/height values will it have? same question, but with Openfl Textfields, are they consistent across platforms?

**Reply by DigiEggz** [reply]
FlxText itself also shares the same problem. I only noticed this because I'm creating UI elements and saw a spacing difference between desktop and web.

For desktop (tested Windows, macOS, Neko), when I set FlxText to 0, 0 there is a noticeable +1px gap from the edge of the screen where the text renders. HTML5 (and Flash) render closer to the origin, where it should be rendered.

I'll try an test later to see if there are any differences for a centered text's width/height values.

**Reply by DigiEggz** [reply]
After testing a plain OpenFL TextField, I believe the issue lies with the TextField itself. The desktop and HTML5 versions render in slightly different locations.

**Reply by Geokureli** [reply]
I would make github issue on OpenFL's issue tracker

**Reply by DigiEggz** [reply]
I created a new [issue](https://github.com/openfl/openfl/issues/2544) for it. Hopefully we can get something going.

**Comment by DigiEggz** [comment, marked as answer]
Fixed by #2536
